packages/bundle/src/index-es5.ts

```diff
@@ -13,8 +16,16 @@ import 'core-js/modules/es6.symbol';
 import 'core-js/modules/es7.array.includes';
 import 'url-search-params-polyfill';
 import 'whatwg-fetch';
-
 import addVersion from './addVersion';
+import defaultCreateDirectLine from './createDirectLine';
+import { version } from './index-minimal';
+
+const createDirectLine = options => defaultCreateDirectLine({ botAgent: `webchat-es5/${version}`, ...options });
```

Comment:
The Web Chat BotAgent is appended to the DirectLine [client agent string](https://github.com/microsoft/BotFramework-DirectLineJS/blob/42a1070a65c4fbeb1ea9ba3f8e1c20aeeb70bc3f/src/directLine.ts#L905) in DirectLineJs, so I don't think there will be an issue with that being overwritten. 

What would be the best approach to adding Web Chat to a custom DirectLine Object? Would we have to check if the DirectLine's BotAgent includes 'webchat' in BasicWebChat or Composer? What if the DirectLine object is a WebChatAdapter and doesn't have the BotAgent property?

Reply:
Two articles I read around the current handling of user agent string in mobile platforms.

- https://www.scientiamobile.com/correctly-form-user-agents-for-mobile-apps/
- https://www.scientiamobile.com/correctly-form-user-agents-for-webview-apps/

In short, setting user agent string is a replacement operation. And the developer is responsible for and should retain the existing user agent string because "is often used by servers to help identify the scope of reported interoperability problems, to work around or tailor responses to avoid particular user agent limitations, and for analytics regarding browser or operating system use" according to the [specification](https://tools.ietf.org/html/rfc7231#section-5.5.3).

The article suggest developers should construct/read the original user agent string and append to it, instead of replacing it. It mentioned Facebook, Twitter, and Instagram is doing the same too.

I think there are two problems here, the first one is easy to solve, the second one need some thinking:

1. When the app call `createDirectLine()`, Web Chat will append bot agent that looks like `WebChat/x.y.z (Full)` to it
   - Dev cannot override the bot agent string
   - When dev pass `botAgent` to `createDirectLine`, we will give a warning and ignore it (just like dev cannot set user agent using `fetch`
1. Devs can optionally set their own bot agent string, they are responsible for and should append `WebChat/x.y.z (Full)` to it

I think #1985 is only about point 1. So let's scope this PR down to 1 first and we figure out 2 later.

If we want to achieve 2, let's make another PR and discuss. Few approaches in my mind:

1. Devs can somehow get/construct Web Chat default user agent string, and they should append it when they set `botAgent`
1. Devs can set `appName` and `appVersion` and we help append to the bot agent
1. Devs set bot agent, but internally in Web Chat, we append to it
   - Personally don't like this one, because it is not how other systems do

What do you think?

Reply:
@tdurnford do you mind filing an issue for devs being able to append their own bot agent string? :)

Reply:
Filled a new issue #2119